### DISCUSSION TITLE
Cluster DNS not working

### DISCUSSION THREAD

**Discussion by jakoberpf** [k3s-io/k3s#9720] [Q&A]
**Environmental Info:**
K3s Version: 
```
k3s version v1.28.7+k3s1 (051b14b2)
go version go1.21.7
```

Node(s) CPU architecture, OS, and Version: 
```
Linux ... 5.15.0-1049-oracle #55~20.04.1-Ubuntu SMP Wed Nov 22 17:38:08 UTC 2023 aarch64 aarch64 aarch64 GNU/Linux
```

Cluster Configuration:
```yaml
servers: 3 (hetzner)
agents: 4 (oracle and strato)
```

All nodes are configured with a wireguard mesh on `10.8.0.0/24` on `wg0`

`server.service`
```cfg
[Unit]
Description=Lightweight Kubernetes
Documentation=https://k3s.io
After=network-online.target

[Service]
Type=notify
ExecStartPre=-/sbin/modprobe br_netfilter
ExecStartPre=-/sbin/modprobe overlay
ExecStart=/usr/local/bin/k3s server --flannel-iface=wg0 --node-ip=10.8.0.XX  --tls-san 123.123.123.123 --disable servicelb --disable traefik --kube-apiserver-arg service-node-port-range=22-32767
KillMode=process
Delegate=yes
# Having non-zero Limit*s causes performance problems due to accounting overhead
# in the kernel. We recommend using cgroups to do container-local accounting.
LimitNOFILE=1048576
LimitNPROC=infinity
LimitCORE=infinity
TasksMax=infinity
TimeoutStartSec=0
Restart=always
RestartSec=5s

[Install]
WantedBy=multi-user.target
```

`agent.service`
```cfg
Description=Lightweight Kubernetes
Documentation=https://k3s.io
After=network-online.target
After=network-online.target

[Install]
WantedBy=multi-user.target

[Service]
Type=notify
EnvironmentFile=-/etc/default/%N
EnvironmentFile=-/etc/sysconfig/%N
EnvironmentFile=-/etc/systemd/system/k3s-agent.service.env
KillMode=process
Delegate=yes
# Having non-zero Limit*s causes performance problems due to accounting overhead
# in the kernel. We recommend using cgroups to do container-local accounting.
LimitNOFILE=1048576
LimitNPROC=infinity
LimitCORE=infinity
TasksMax=infinity
TimeoutStartSec=0
Restart=always
RestartSec=5s
ExecStartPre=/bin/sh -xc '! /usr/bin/systemctl is-enabled --quiet nm-cloud-setup.service'
ExecStartPre=-/sbin/modprobe br_netfilter
ExecStartPre=-/sbin/modprobe overlay
ExecStart=/usr/local/bin/k3s agent --server https://1123.123.123.123:6443 --token ... --flannel-iface=wg0 --node-ip=10.8.0.XX
```

**Describe the bug:**
The cluster dns is not working, so I get errors like `transport: Error while dialing: dial tcp: lookup istiod.istio-system.svc on 10.43.0.10:53: read udp 10.42.3.3:43034->10.43.0.10:53: i/o timeout`. I tested setup with `ufw` and `53` open and also with `ufw`disabled. No I don't know how to further troubleshoot the issue. The firewalls are configured do allow all udp and tcp. `coredns` looks fine to me.

And tipps how do debug this would be much appreciated...

**Comment by jakoberpf** [comment]
I am looking to this https://github.com/k3s-io/k3s/issues/535#issuecomment-907789701

**Comment by brandond** [comment]
> servers: 3 (hetzner)
> agents: 4 (oracle and strato)
> All nodes are configured with a wireguard mesh on 10.8.0.0/24 on wg0

The behavior you are describing indicates that UDP traffic between cluster members is being dropped. You might see if you need to reduce the MTU in order to fit the encapsulated VXLAN traffic through the wireguard tunnel.

You might also try using the built-in wireguard/tailscale support, to avoid double-encapsulating your traffic:
https://docs.k3s.io/installation/network-options#integration-with-the-tailscale-vpn-provider-experimental

I'm going to convert this to a discussion, as it appears to be a configuration issue, not a defect in k3s.
